Code Review for DFPK Pi ZeroW

bertokhoury opened the pull request:
The page can be accessed at 192.168.42.1/cgi-bin/dfpk-wifi

Later this page will become a captive portal by acting as a FAS for
nodogsplash

MasonAsh committed:
Create wifi connection page

The page can be accessed at 192.168.42.1/cgi-bin/dfpk-wifi

Later this page will become a captive portal by acting as a FAS for
nodogsplash

ianblenke commented on Dockerfile:
```diff
@@ -67,7 +67,7 @@ ENV PACKAGES=git-http
 WORKDIR /imagebuilder
 
-ENV PACKAGES="${PACKAGES} avahi-daemon avahi-daemon-service-ssh avahi-daemon-service-http base-files bash blkid block-mount busybox curl cryptsetup dnsmasq dropbear ca-bundle ca-certificates e2fsprogs f2fs-tools firewall fstools gcc git ip6tables iptables iw i2c-tools jshn jsonfilter kernel kmod-ath kmod-ath9k-common kmod-cfg80211 kmod-crypto-xts kmod-crypto-iv kmod-crypto-misc kmod-crypto-user k ...
+ENV PACKAGES="${PACKAGES} avahi-daemon avahi-daemon-service-ssh avahi-daemon-service-http base-files bash blkid block-mount busybox curl cryptsetup dnsmasq dropbear ca-bundle ca-certificates e2fsprogs f2fs-tools firewall fstools gcc git ip6tables iptables iw i2c-tools jshn jsonfilter kernel kmod-ath kmod-ath9k-common kmod-cfg80211 kmod-crypto-xts kmod-crypto-iv kmod-crypto-misc kmod-crypto-user k ...
```
I get the `nodogsplash`, but do we need the `openssh-sftp-server` for anything, or is that just a debugging addition?

bertokhoury replied:
Yeah that was just to be able to shh into it.

ianblenke replied:
Any more improvements to this?

MasonAsh replied:
I've added a page to fetch and apply the tor config from the REST api. The next steps are to allow the config to be fetched from an onion address and then to setup the `dfpk-wifi` page as a forward authenticating service for `nodogsplash`.

ianblenke replied:
I see some changes to use tor as a socks5 proxy... is this ready for me to try yet?

bertokhoury replied:
Yes, socks5 proxy is good to go.

ianblenke reviewed (changes requested):
It looks like the initial wifi configuration stuff as a captive portal is here. I'm guessing that the tor configuration stuff is still under development.

MasonAsh committed:
Add a page to connect to private tor network REST

The page will write to a custom config file located at:

/etc/config/dfpk-torrc

The tor service script was modified to load this config file if it
exists.

MasonAsh committed:
Configure nodogsplash to redirect to wifi page

MasonAsh committed:
Allow connection to a REST api on a hidden service.

MasonAsh committed:
Remove unused variable

ianblenke commented:
Merging and taking it from here. Merry Christmas!
